Library/Homebrew/cask/dsl.rb

```diff
@@ -525,12 +531,25 @@ def livecheckable?
     # NOTE: A warning will be shown when trying to install this cask.
     #
     # @api public
-    def deprecate!(date:, because:, replacement: nil)
+    def deprecate!(date:, because:, replacement: nil, replacement_formula: nil, replacement_cask: nil)
+      if replacement_formula && replacement_cask
+        raise ArgumentError, "replacement_formula and replacement_cask specified!"
+      end
+
+      if replacement
+        odeprecated(
+          "deprecate!(:replacement)",
+          "deprecate!(:replacement_formula) or deprecate!(:replacement_cask)",
+          disable_on: Time.new(2025, 10, 15),
+        )
+      end
+
       @deprecation_date = Date.parse(date)
       return if @deprecation_date > Date.today
 
       @deprecation_reason = because
       @deprecation_replacement = replacement
+      @deprecation_replacement_type = replacement_type
```

Comment:
Feels like here (and below) we should either set:
- `@(deprecation|disable)_replacement` and `@(deprecation|disable)_replacement_type`
or:
- `@(deprecation|disable)_replacement_formula` and `@(deprecation|disable)_replacement_cask`
but not both.

The `deprecate!` and `disable!` public APIs need to remain backwards compatible but the internal `DSL_METHODS` and `attr_reader`s do not.